### PR TITLE
Prepare all necessary JS artifacts for build task

### PR DESCRIPTION
**Pull request by alexslavr**

### CODE REVIEW

File: .github/workflows/update_version.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      version:
+        type: string
+        description: XX.X.X Set exact version (optional)
+        required: true
+        default: 23.2.1
+
+jobs:
+  update_version:
+    name: Update version and create PR
+    runs-on: ubuntu-latest
```
Comment (timbset) [inline .github/workflows/update_version.yml:18]: Don't we need to use self-hosted runner?

File: build/build-all.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+const MONOREPO_ROOT = path.join(__dirname, '..');
+const INTERNAL_TOOLS_ARTIFACTS = path.join(MONOREPO_ROOT, 'artifacts', 'internal-tools');
+
+const OUTPUT_DIR = path.join(MONOREPO_ROOT, 'artifacts');
+const NPM_OUTPUT_DIR = path.join(OUTPUT_DIR, 'npm');
+
+sh.mkdir('-p', NPM_OUTPUT_DIR);
+
+const { "devextreme-main": devextremeVersion, devextreme: devextremeNpmVersion } = JSON.parse(sh.exec('npm pkg get version -ws --json').stdout);
```
Comment (timbset) [inline build/build-all.ts]:
```suggestion
const { 'devextreme-main': devextremeVersion, devextreme: devextremeNpmVersion } = JSON.parse(sh.exec('npm pkg get version -ws --json').stdout);
```

File: build/build-all.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import sh from "shelljs";
+import path from "node:path";
```
Comment (timbset) [inline build/build-all.ts]:
```suggestion
import sh from 'shelljs';
import path from 'node:path';
```

File: .github/workflows/update_version.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+name: Update version in all workspaces
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch_name:
+        type: string
+        description: XX_X (23_2+)
+        required: true
+        default: 23_2
```
Comment (IlyaKhD) [inline .github/workflows/update_version.yml]: I would avoid adding defaults, if possible

File: build/update-version.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import sh from "shelljs";
+import path from "node:path";
```
Comment (timbset) [inline build/update-version.ts]:
```suggestion
import sh from 'shelljs';
import path from 'node:path';
```

**Review by timbset — CHANGES_REQUESTED** [review]

File: build/build-all.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+sh.popd();
+
+// TODO: maybe we should add bootstrap to vendors
+const BOOTSTRAP_DIR = path.join(MONOREPO_ROOT, 'node_modules', 'bootstrap', 'dist');
+sh.cp([path.join(BOOTSTRAP_DIR, 'js', 'bootstrap.js'), path.join(BOOTSTRAP_DIR, 'js', 'bootstrap.min.js')], path.join(OUTPUT_DIR, 'js'));
+sh.cp([path.join(BOOTSTRAP_DIR, 'css', 'bootstrap.css'), path.join(BOOTSTRAP_DIR, 'js', 'bootstrap.min.css')], path.join(OUTPUT_DIR, 'css'));
+
+// Update versions for non-semver builds (daily, alpha and beta)
+if (devextremeVersion !== devextremeNpmVersion) {
+    sh.exec(`npm run all:bump -- ${devextremeNpmVersion}`);
```
Comment (IlyaKhD) [inline build/build-all.ts]: Typo?
```suggestion
    sh.exec(`npm run all:update-version -- ${devextremeNpmVersion}`);
```

File: .github/workflows/update_version.yml (hunk 1 above)
Comment (Lykoi18) [reply .github/workflows/update_version.yml:18]: In this action uses the GitHub CLI [command](https://github.com/DevExpress/DevExtreme/pull/25842/files/52ff15a9b1c7f6d8ec14928f5c12ba5ad6244f12#diff-8d4c99794a132146cc00f3b4860b4896f1627711754c64088d1005786ab3f0d9R53), but it is not installed in the self-hosted runner, so we use github-hosted runner.

**Review by timbset — APPROVED** [review]

**Pull request merged**